front/src/modules/favorites/components/Favorites.tsx

```diff
@@ -1,59 +1,100 @@
+import { useEffect, useState } from 'react';
+import React from 'react';
 import styled from '@emotion/styled';
+import { Reorder } from 'framer-motion';
 
 import NavItem from '@/ui/navbar/components/NavItem';
 import NavTitle from '@/ui/navbar/components/NavTitle';
 import { Avatar } from '@/users/components/Avatar';
 import { useGetFavoritesQuery } from '~/generated/graphql';
 import { getLogoUrlFromDomainName } from '~/utils';
 
+import { useFavorites } from '../hooks/useFavorites';
+
 const StyledContainer = styled.div`
   display: flex;
   flex-direction: column;
+  height: 100%;
   overflow-x: auto;
   width: 100%;
 `;
 
 export const Favorites = () => {
   const { data } = useGetFavoritesQuery();
-  const favorites = data?.findFavorites;
+  const { updateFavoritesOrder } = useFavorites();
+
+  const [favorites, setFavorites] = useState(data?.findFavorites);
+
+  useEffect(() => {
+    const _favorites = data?.findFavorites;
+    const _finalFav = _favorites ? Array.from(_favorites) : [];
+    setFavorites(_finalFav);
+  }, [data?.findFavorites]);
```

Comment:
Hi @abhithory, thank you for contributing!

You don't need a `useEffect`, instead you can use the `onCompleted` callback like this:

```typescript
  const [favorites, setFavorites] = useState<
    GetFavoritesQuery['findFavorites']
  >([]);

  useGetFavoritesQuery({
    onCompleted: (data) => setFavorites(data?.findFavorites ?? []),
  });
```

Reply:
ok,